pilot/pkg/model/sidecar.go

```diff
@@ -280,20 +287,33 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
 		Namespace: sidecarConfig.Namespace,
 	}.HashCode())
 
-	egressConfigs := sidecar.Egress
+	if !features.EnableLazySidecarEvaluation {
+		initSidecarScopeInternalIndexes(ps, out, configNamespace)
+	} else {
+		out.initFunc = sync.OnceFunc(func() {
```

Comment:
Can we just call so that we do not have to call initFunc seperately in tests and what we execute is same when EnableLazySidecarEvaluation is enabled/disabled?

	if !features.EnableLazySidecarEvaluation {
		out.initFunc()
	}

Reply:
Sure, I can do that I was just trying to avoid adding any overhead when flag is disabled. Regarding tests, I still need to call `initFunc` in some tests because I need to make sure SidecarScopes are initialized to be able to compare them.

Reply:
@ramaraochavali I pushed some changes, let me know if they seem ok